Assets/Mirror/Components/Discovery/NetworkDiscoveryHUD.cs

```diff
@@ -17,11 +18,20 @@ public class NetworkDiscoveryHUD : MonoBehaviour
 #if UNITY_EDITOR
         void OnValidate()
         {
-            if (networkDiscovery == null)
+            if (Application.isPlaying) return;
+            Reset();
+        }
+
+        void Reset()
+        {
+            networkDiscovery = GetComponent<NetworkDiscovery>();
+
+            // Add default event handler if not already present
+            if (!Enumerable.Range(0, networkDiscovery.OnServerFound.GetPersistentEventCount())
+                .Any(i => networkDiscovery.OnServerFound.GetPersistentMethodName(i) == "OnDiscoveredServer"))
```

Comment:
is there any cleaner way for this? 
- no string search?
- maybe networkDiscovery.OnServerFound -= OnDiscoveredServer to remove old if any, and then += to add it?

Reply:
I can change it to `nameof(OnDiscoveredServer)` I think, but remove and add doesn't show in the inspector

![image](https://github.com/user-attachments/assets/fdadd0ba-7d8d-496d-980c-bd8210f650b2)